docs/learning/WriteAConnector.md

```diff
@@ -1,108 +1,727 @@
+
+In its [ImportDefinitions](#importdefinitions) and [UpdateExistingData](#updateexistingdata) functions, the connector must do the following.
+
+For each item found in the external source.
+
+Define a [[SourceItem]] object to capture the identifier, version, and checksum of the item.
+
+```JavaScript
+    const sourceItem: SourceItem = {
+      id: <<identifies the item in the external source>>,
+      version: <<the item’s version number, if available>>,
+      checksum: <<the checksum of the item’s content>>,
+    };
 ```
 
-## Detecting and pushing changes
+Then, check use [Synchronizer](#synchronizer) to see if the item was previously converted.
+
+```JavaScript
+const results = this.synchronizer.detectChanges(scopeId, kind, sourceItem);
+```
+
+If so, check to see if the item’s state is unchanged since the previous conversion. In that case, register the fact that the element in the iModel is still required and move on to the next item.
+
+```JavaScript
+    if (results.state === ItemState.Unchanged) {
+      assert(results.id !== undefined);
+      this.synchronizer.onElementSeen(results.id);
+      return results.id;
+    }
+```
+
+Otherwise, the item is new or if its state has changed. The connector must generate the appropriate BIS element representation of it.
+
+```JavaScript
+    const element = convert the item to BIS element …
+
+    if (results.id !== undefined) // in case this is an update
+      element.id = results.id;
+```
+
+Then ask Synchronizer to write the BIS element to the briefcase. Synchronizer will update the existing element if it exists or insert a new one if not.
+
+```JavaScript
+    const sync: SynchronizationResults = {
+      element,
+      itemState: results.state,
+    };
+    const status = this.synchronizer.updateIModel(sync, scopeId, sourceItem, kind);
+
+    if (status !== IModelStatus.Success)
+      throw new Error(`Failed to write ${sourceItem.id}. Status=${status}`);
+
+    return sync.element.id;
+
+}
+```
+
+Note that Synchronizer.updateIModel automatically adds an [[ExternalSourceAspect]] to the element, to keep track of the mapping between it and the external item.
+
+The framework will automatically detect and delete elements and models if the corresponding external items were not updated or declared as seen.
+
+## Connector SDK
+
+### Getting started
+
+You'll need Node.js version ">=12.17.0 <15.0”. Please refer to [Section 1 from iTwin.js Getting Started](https://www.itwinjs.org/getting-started/) for more details.
+
+The node packages you'll need can be installed using
+
+```Shell
+$npm install  @bentley/backend-itwin-client
+$npm install  @bentley/bentleyjs-core
+$npm install  @bentley/context-registry-client
+$npm install  @bentley/geometry-core
+$npm install  @bentley/imodelhub-client
+$npm install  @bentley/imodeljs-backend
+$npm install  @bentley/imodeljs-common
+$npm install  @bentley/rbac-client
+
+$npm install  --save-dev @bentley/backend-itwin-client

```

Comment:
A handful of these should be real deps or are duplicated from the list above.

The list should be;

```sh
$npm install  @bentley/backend-itwin-client
$npm install  @bentley/bentleyjs-core
$npm install  @bentley/context-registry-client
$npm install  @bentley/ecschema-metadata
$npm install  @bentley/geometry-core
$npm install  @bentley/imodelhub-client
$npm install  @bentley/imodeljs-backend
$npm install  @bentley/imodeljs-common
$npm install  @bentley/itwin-client
$npm install  @bentley/rbac-client
$npm install  @bentley/telemetry-client

$npm install  --save-dev @bentley/build-tools
$npm install  --save-dev @bentley/config-loader
$npm install  --save-dev @bentley/eslint-plugin
$npm install  --save-dev @bentley/oidc-signin-tool

$npm install  --save-dev chai
$npm install  --save-dev cpx
$npm install  --save-dev eslint
$npm install  --save-dev mocha
$npm install  --save-dev nyc
$npm install  --save-dev rimraf
$npm install  --save-dev typescript
```